src/controller.cpp

```diff
@@ -752,6 +729,48 @@ QList<IPAddressRange> Controller::getAllowedIPAddressRanges(
   return list;
 }
 
+QStringList Controller::getExcludedAddresses(const QList<Server>& serverList) {
+  logger.debug() << "Computing the excluded IP addresses";
+
+  QStringList list;
+
+  // filtering out the captive portal endpoint
+  if (FeatureCaptivePortal::instance()->isSupported() &&
+      SettingsHolder::instance()->captivePortalAlert()) {
+    CaptivePortal* captivePortal = MozillaVPN::instance()->captivePortal();
+
+    for (const QString& address : captivePortal->ipv4Addresses()) {
+      logger.debug() << "Filtering out the captive portal address:" << address;
+      list.append(address);
+    }
+    for (const QString& address : captivePortal->ipv6Addresses()) {
+      logger.debug() << "Filtering out the captive portal address:" << address;
+      list.append(address);
+    }
+  }
+
+  // Filter out the Custom DNS Server, if the user has set one.
+  if (DNSHelper::shouldExcludeDNS()) {
+    auto dns = DNSHelper::getDNS(serverList.last().ipv4Gateway());
+    logger.debug() << "Filtering out the DNS address:" << dns;
+    list.append(dns);
+  }
+
+  // It's not clear if this is really needed for the Android platform, but
```

Comment:
Should also not be required for android, as far as i remember

Reply:
Awesome thanks!